src/notebooks/controllers/installPythonControllerCommands.ts

```diff
@@ -161,4 +176,40 @@ export class InstallPythonControllerCommands implements IExtensionSingleActivati
             }
         }
     }
+
+    // We don't always want to show our modal warning for installing the python extension
+    // this function will choose if this should be shown, and return true if the install should
+    // proceed and false otherwise
+    private async shouldInstallExtensionPrompt(): Promise<boolean> {
+        // We want to show the dialog if the active document is running, in this case, the command
+        // was triggered from the run button and we want to warn the user what we are doing
+        if (this.isActiveNotebookDocumentRunning()) {
```

Comment:
Put some hefty comments here, as this is "cuter" than I generally like to get. Per feedback from Jim on the v-team this change means that you only get the model "Sure you want to install extension" dialog when you use the run button to trigger the command. If you just directly click it in the kernel picker you don't get the modal dialog.